harmony/harmonydb/sql/20241017-market-mig-indexing.sql

```diff
@@ -0,0 +1,88 @@
+-- All indexing task entries are made into this table
+-- and then copied over to market_mk12_deal_pipeline for a controlled migration
+CREATE TABLE market_mk12_deal_pipeline_migration (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    sp_id BIGINT NOT NULL,
+    piece_cid TEXT NOT NULL,
+    piece_size BIGINT NOT NULL, -- padded size
+    raw_size BIGINT DEFAULT NULL,
+    sector BIGINT DEFAULT NULL,
+    reg_seal_proof INT DEFAULT NULL,
+    sector_offset BIGINT DEFAULT NULL, -- padded offset
+    should_announce BOOLEAN NOT NULL
+);
+
+CREATE OR REPLACE FUNCTION migrate_deal_pipeline_entries()
+RETURNS VOID
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    -- Counts for existing entries and tasks
+    num_entries_ready_for_indexing INTEGER;
+    num_pending_ipni_tasks INTEGER;
+    num_entries_needed INTEGER;
+    num_entries_to_move INTEGER;
+    cnt INTEGER;
+    moved_rows RECORD;
+BEGIN
+    -- Step 1: Check if the migration table has entries
+    SELECT COUNT(*) INTO cnt FROM market_mk12_deal_pipeline_migration;
```

Comment:
Probably LIMIT here and in other count selects

Reply:
Done with 16 Limit as we don't need more anyway.